File: backend/app/schemas/stock.py
from pydantic import BaseModel
from typing import List, Optional

class StockSearchResult(BaseModel):
    symbol: str
    name: str
    type: str
    exchange: str

class StockSearch(BaseModel):
    query: str
    results: List[StockSearchResult]

class StockInfo(BaseModel):
    symbol: str
    name: str
    currency: str
    exchange: str
    sector: str
    industry: str
    marketCap: float
    website: str
    description: str
    employees: int
    country: str
    currentPrice: float
    previousClose: float
    open: float
    dayLow: float
    dayHigh: float
    fiftyTwoWeekLow: float
    fiftyTwoWeekHigh: float
    volume: int
    averageVolume: int
    dividendYield: Optional[float] = 0
    beta: Optional[float] = 0
    trailingPE: Optional[float] = 0
    forwardPE: Optional[float] = 0

class HistoricalData(BaseModel):
    date: str
    open: float
    high: float
    low: float
    close: float
    volume: int

class StockHistory(BaseModel):
    symbol: str
    period: str
    data: List[HistoricalData]

class StockQuote(BaseModel):
    symbol: str
    name: str
    price: float
    change: float
    changePercent: float
    previousClose: float
    open: float
    dayLow: float
    dayHigh: float
    volume: int
    marketCap: float
    timestamp: str

File: frontend/src/types/stock.ts
export interface StockInfo {
  symbol: string
  name: string
  currency: string
  exchange: string
  sector: string
  industry: string
  marketCap: number
  website: string
  description: string
  employees: number
  country: string
  currentPrice: number
  previousClose: number
  open: number
  dayLow: number
  dayHigh: number
  fiftyTwoWeekLow: number
  fiftyTwoWeekHigh: number
  volume: number
  averageVolume: number
  dividendYield?: number
  beta?: number
  trailingPE?: number
  forwardPE?: number
}

export interface HistoricalData {
  date: string
  open: number
  high: number
  low: number
  close: number
  volume: number
}

export interface StockHistory {
  symbol: string
  period: string
  data: HistoricalData[]
}

export interface StockQuote {
  symbol: string
  name: string
  price: number
  change: number
  changePercent: number
  previousClose: number
  open: number
  dayLow: number
  dayHigh: number
  volume: number
  marketCap: number
  timestamp: string
}
